### PR TITLE
Rip out savegame compatibility code

### PR DESCRIPTION
**Pull request by gunchleoc**
@Noordfrees I have 1 NOCOM comment in there where I wasn't sure how to proceed. Could you take care of that one? Whenever you find the time :)

### CODE REVIEW

File: src/economy/shipping_schedule.cc [hunk 1]
```diff
@@ -1628,6 +1628,8 @@ void ShippingSchedule::save(const EditorGameBase& egbase,
 
 		// TODO(Nordfriese): Replace with at() when we break savegame compatibility
 		// (can only be not-present in compatibility cases)
+		// NOCOM I'm not entirely sure what you intended here,
+		// could you please take care of this one?
 		auto it = last_actual_duration_recalculation_.find(pair.first);
 		fw.unsigned_32(it == last_actual_duration_recalculation_.end() ? 0 : it->second);
```
Comment (Noordfrees) [inline src/economy/shipping_schedule.cc]:
```suggestion
		fw.unsigned_32(last_actual_duration_recalculation_.at(pair.first));
```

File: src/logic/queue_cmd_ids.h [hunk 2]
```diff
@@ -61,9 +61,9 @@ enum class QueueCommandTypes : uint8_t {
 
 	kSetWarePriority,
 	kSetWareTargetQuantity,
-	kResetWareTargetQuantity,
-	kSetWorkerTargetQuantity,
-	kResetWorkerTargetQuantity,  // 16
+	// 14 removed post Build 21
+	kSetWorkerTargetQuantity,  // 15
```
Comment (Noordfrees) [inline src/logic/queue_cmd_ids.h]: doesn't this need an `= 15`?

Comment (gunchleoc) [reply src/logic/queue_cmd_ids.h]: It does, thanks.

**Review by Noordfrees — COMMENTED** [review]
src/economy/economy_data_packet.cc line 36 still contains compat code

**Comment by gunchleoc** [comment]
> src/economy/economy_data_packet.cc line 36 still contains compat code

Thanks - fixed.

**Review by Noordfrees — APPROVED** [review]

**Comment by gunchleoc** [comment]
Transitory Appveyor failure, will merge.

**Pull request merged**